src/renderer/routes/flow_chart/FlowChartTabView.tsx

```diff
@@ -579,9 +582,10 @@ const FlowChartTab = () => {
             onInit={onInit}
             onNodesChange={onNodesChange}
             onEdgesChange={onEdgesChange}
-            onConnect={onConnect}
+            onConnect={withPermissionCheck(onConnect)}
+            nodesDraggable={isAdmin()}
             onNodeDragStop={handleNodeDrag}
-            onNodesDelete={handleNodesDelete}
+            onNodesDelete={withPermissionCheck(handleNodesDelete)}
```

Comment:
This isn't quite what I had in mind..
What I was thinking of was maybe getting rid of `useFlowChartGraph` and doing some refactoring so that `setNodes/setEdges` would only be exposed through an hook that wraps it with the permission check:
```
export const useSetNodes = () => withPermissionCheck(useSetAtom(nodesAtom));
```
So that anything that requires `setNodes` would have to go through this check, making it impossible to accidentally modify the nodes outside of that file without doing the permission check
Anything that requires setting the nodes without needing editing permissions (e.g loading a file) would have to explicitly define a hook/function in that file so that it would have direct access to the underlying atom.

The way you did it is explicit whenever something requires permissions, which isn't a bad thing, but it can be annoying to write out the auth check each time, really it's a thing of opt in (your implementation) vs opt out (my suggestion)

I'm curious to hear your thoughts on this.